tests/foreman/api/test_lifecycleenvironment.py

```diff
@@ -24,204 +24,203 @@
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+@tier1
+def test_positive_create_with_name(name):
+    """Create lifecycle environment with valid name only
 
-        :CaseImportance: Critical
-        """
-        for name in valid_data_list():
-            with self.subTest(name):
-                lc_env = entities.LifecycleEnvironment(organization=self.org, name=name).create()
-                self.assertEqual(lc_env.name, name)
+    :id: ec1d985a-6a39-4de6-b635-c803ecedd832
 
-    @tier1
-    def test_positive_create_with_description(self):
-        """Create lifecycle environment with valid description
+    :expectedresults: Lifecycle environment is created and has proper name
 
-        :id: 0bc05510-afc7-4087-ab75-1065ab5ba1d3
+    :CaseImportance: Critical
+    """
+    assert entities.LifecycleEnvironment(name=name).create().name == name
 
-        :expectedresults: Lifecycle environment is created and has proper
-            description
-
-        :CaseImportance: Critical
-        """
-        description = gen_string('utf8')
-        lc_env = entities.LifecycleEnvironment(
-            organization=self.org, description=description
-        ).create()
-        self.assertEqual(lc_env.description, description)
 
-    @tier1
-    def test_positive_create_prior(self):
-        """Create lifecycle environment with valid name, prior to
-        Library
+@pytest.mark.parametrize('desc', **parametrized(valid_data_list()))
+@tier1
+def test_positive_create_with_description(desc):
+    """Create lifecycle environment with valid description
 
-        :id: 66d34781-8210-4282-8b5e-4be811d5c756
+    :id: 0bc05510-afc7-4087-ab75-1065ab5ba1d3
 
-        :expectedresults: Lifecycle environment is created with Library as
-            prior
+    :expectedresults: Lifecycle environment is created and has proper
+        description
 
-        :CaseImportance: Critical
-        """
-        lc_env = entities.LifecycleEnvironment(organization=self.org).create()
-        self.assertEqual(lc_env.prior.read().name, ENVIRONMENT)
+    :CaseImportance: Critical
+    """
+    assert entities.LifecycleEnvironment(description=desc).create().description == desc
 
-    @tier3
-    def test_negative_create_with_invalid_name(self):
-        """Create lifecycle environment providing an invalid name
 
-        :id: 7e8ea2e6-5927-4e86-8ea8-04c3feb524a6
+@tier1
+def test_positive_create_prior(module_org):
+    """Create lifecycle environment with valid name, prior to
+    Library
 
-        :expectedresults: Lifecycle environment is not created
+    :id: 66d34781-8210-4282-8b5e-4be811d5c756
 
-        :CaseImportance: Low
-        """
-        for name in invalid_values_list():
-            with self.subTest(name):
-                with self.assertRaises(HTTPError):
-                    entities.LifecycleEnvironment(name=name).create()
+    :expectedresults: Lifecycle environment is created with Library as
+        prior
```

Comment:
Aren't these descriptions inconsistent?

If something is "prior to Library", then it's first "something" and then Library.

If something is created with "Library as prior", then it's first Library, and then "something".

Or is there some language nuance that I am missing?

Reply:
It's the latter.  "Library as prior" as you've stated so it's first Library then "something"

Reply:
Could you please change first line in dosctring, so it's consistent with expectedresults and your comment?